coopaname_custom/models/event.py

```diff
@@ -102,7 +103,7 @@ def sync_co_organizer_registrations(self, new_co_organizer_id):
             new_registration.confirm_registration()
 
     @api.model
-    def create(self, vals):
+    def create(self, vals) -> Event:
```

Comment:
It actually returns a RecordSet of model `event.event` rather than Event class.

Reply:
Oh thanks that's totally right will fix it.

Reply:
Looked into odoo source code and doc it look like that `RecordSet` is not explicitly defined as a class it's just an odered collection of something who inherit `model.Model` if I am correct the docs.

In Odoo stubs the author choses [this signature](https://github.com/trinhanhngoc/odoo-stubs/blob/171cb8f0c2d9427bfe4e5b8d7a5627afc0546349/odoo-stubs/models.pyi#L199)

Citation from <https://github.com/trinhanhngoc/odoo-stubs/> by trinhanhngoc
```python
    def create(self, vals_list: Any) -> BaseModel: ...
```

Since every class who inherit BaseModel behave like and are a `RecordSet` 

And about `event.event` is defined here: <https://github.com/OCA/event> but didn't find real class behind so maybe leaving out the «interface» here `BaseModel` is better than nothing, except you have some insight will need some check at runtime to get the right type. 

See what you said with types and odoo, not impossible but involve some corners case due to ultra dynamic object gym.